src/main/java/taxi/controller/AddDriverController.java

```diff
@@ -27,7 +27,12 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         String licenceNumber = req.getParameter("licenceNumber");
         String login = req.getParameter("login");
         String password = req.getParameter("password");
-        Driver driver = driverService.create(new Driver(name,licenceNumber,login,password));
-        resp.sendRedirect(req.getContextPath() + "/drivers/all");
+        String repeatPassword = req.getParameter("repeat_password");
+        if (password.equals(repeatPassword)) {
+            driverService.create(new Driver(name, licenceNumber, login, password));
+            resp.sendRedirect(req.getContextPath() + "/drivers/login");
+        }
+        req.setAttribute("error Message", "Passwords are not identical!");
+        req.getRequestDispatcher("/WEB-INF/views/drivers/login");
```

Comment:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/drivers/login");
```
This won't work without .forward() I think